GITHUB_PR_NUMBER not defined

I configured as instructed to build heads, and got:

```
13:09:47 GitHub PR #<a href="https://git.basistech.net/raas/rosapi1.5/pull/273">273</a>, PR opened
13:09:47 Building in workspace /var/jenkins_home/workspace/rosapi1.5-pull-requests
13:09:47  > git rev-parse --is-inside-work-tree # timeout=10
13:09:47 Fetching changes from the remote Git repository
13:09:47  > git config remote.origin-pull.url git@git.basistech.net:raas/rosapi1.5.git # timeout=10
13:09:47 Fetching upstream changes from git@git.basistech.net:raas/rosapi1.5.git
13:09:47  > git --version # timeout=10
13:09:47 using GIT_SSH to set credentials 
13:09:47  > git fetch --tags --progress git@git.basistech.net:raas/rosapi1.5.git +refs/pull/${GITHUB_PR_NUMBER}/head:refs/remotes/origin-pull/pull/${GITHUB_PR_NUMBER}/head
13:09:48 ERROR: Error fetching remote repo 'origin-pull'
13:09:48 hudson.plugins.git.GitException: Failed to fetch from git@git.basistech.net:raas/rosapi1.5.git
13:09:48    at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:797)
13:09:48    at hudson.plugins.git.GitSCM.retrieveChanges(GitSCM.java:1051)
13:09:48    at hudson.plugins.git.GitSCM.checkout(GitSCM.java:1082)
13:09:48    at hudson.scm.SCM.checkout(SCM.java:495)
13:09:48    at hudson.model.AbstractProject.checkout(AbstractProject.java:1278)
13:09:48    at hudson.model.AbstractBuild$AbstractBuildExecution.defaultCheckout(AbstractBuild.java:604)
13:09:48    at jenkins.scm.SCMCheckoutStrategy.checkout(SCMCheckoutStrategy.java:86)
13:09:48    at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:529)
13:09:48    at hudson.model.Run.execute(Run.java:1720)
13:09:48    at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
13:09:48    at hudson.model.ResourceController.execute(ResourceController.java:98)
13:09:48    at hudson.model.Executor.run(Executor.java:404)
13:09:48 Caused by: hudson.plugins.git.GitException: Command "git fetch --tags --progress git@git.basistech.net:raas/rosapi1.5.git +refs/pull/${GITHUB_PR_NUMBER}/head:refs/remotes/origin-pull/pull/${GITHUB_PR_NUMBER}/head" returned status code 128:
13:09:48 stdout: 
13:09:48 stderr: fatal: Couldn't find remote ref refs/pull/${GITHUB_PR_NUMBER}/head
13:09:48 
13:09:48    at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1752)
13:09:48    at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandWithCredentials(CliGitAPIImpl.java:1495)
13:09:48    at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.access$300(CliGitAPIImpl.java:64)
13:09:48    at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$1.execute(CliGitAPIImpl.java:315)
13:09:48    at hudson.plugins.git.GitSCM.fetchFrom(GitSCM.java:795)
13:09:48    ... 11 more
13:09:48 ERROR: null
13:09:48 Finished: FAILURE
```

--- a/github-pullrequest-plugin/src/main/java/com/github/kostyasha/github/integration/branch/trigger/JobRunnerForBranchCause.java
+++ b/github-pullrequest-plugin/src/main/java/com/github/kostyasha/github/integration/branch/trigger/JobRunnerForBranchCause.java
@@ -19,7 +19,12 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collection;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 import static com.cloudbees.jenkins.GitHubWebHook.getJenkinsInstance;
 import static com.github.kostyasha.github.integration.branch.data.GitHubBranchEnv.CAUSE_SKIP;
@@ -111,8 +116,9 @@ public class JobRunnerForBranchCause implements Predicate<GitHubBranchCause> {
     }
 
     private QueueTaskFuture<?> startJob(GitHubBranchCause cause) {
-        List<ParameterValue> values = getDefaultParametersValues(job);
-        values.addAll(asList(
+        ParametersAction parametersAction;
+        List<ParameterValue> parameters = getDefaultParametersValues(job);
+        final List<ParameterValue> pluginParameters = asList(
                 //GitHubBranchEnv
                 NAME.param(cause.getBranchName()),
                 SHORT_DESC.param(cause.getShortDescription()),
@@ -124,14 +130,27 @@ public class JobRunnerForBranchCause implements Predicate<GitHubBranchCause> {
                 //GitHubRepoEnv
                 GIT_URL.param(cause.getGitUrl()),
                 SSH_URL.param(cause.getSshUrl())
-        ));
+        );
+        parameters.addAll(pluginParameters);
+
+        try {
+            Constructor<ParametersAction> constructor = ParametersAction.class.getConstructor(List.class, Collection.class);
+            Set<String> names = new HashSet<>();
+            for (ParameterValue param : parameters) {
+                names.add(param.getName());
+            }
+            parametersAction = constructor.newInstance(parameters, names);
+        } catch (NoSuchMethodException | IllegalAccessException | InstantiationException
+                | InvocationTargetException ex) {
+            parametersAction = new ParametersAction(parameters);
+        }
 
         GitHubBranchBadgeAction gitHubBadgeAction = new GitHubBranchBadgeAction(cause);
 
         //TODO no way to get quietPeriod, so temporary ignore it
         return asParameterizedJobMixIn(job).scheduleBuild2(0,
                 new CauseAction(cause),
-                new ParametersAction(values),
+                parametersAction,
                 gitHubBadgeAction);
     }
 

--- a/github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/JobRunnerForCause.java
+++ b/github-pullrequest-plugin/src/main/java/org/jenkinsci/plugins/github/pullrequest/trigger/JobRunnerForCause.java
@@ -32,8 +32,12 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
 import java.util.Collection;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Set;
 
 import static com.cloudbees.jenkins.GitHubWebHook.getJenkinsInstance;
 import static com.google.common.base.Predicates.instanceOf;
@@ -257,8 +261,9 @@ public class JobRunnerForCause implements Predicate<GitHubPRCause> {
     }
 
     private QueueTaskFuture<?> startJob(GitHubPRCause cause) {
-        List<ParameterValue> values = getDefaultParametersValues(job);
-        values.addAll(asList(
+        ParametersAction parametersAction;
+        List<ParameterValue> parameters = getDefaultParametersValues(job);
+        final List<ParameterValue> pluginParameters = asList(
                 TRIGGER_SENDER_AUTHOR.param(cause.getTriggerSenderName()),
                 TRIGGER_SENDER_EMAIL.param(cause.getTriggerSenderEmail()),
                 COMMIT_AUTHOR_NAME.param(cause.getCommitAuthorName()),
@@ -275,7 +280,21 @@ public class JobRunnerForCause implements Predicate<GitHubPRCause> {
                 CAUSE_SKIP.param(cause.isSkip()),
                 NUMBER.param(String.valueOf(cause.getNumber())),
                 STATE.param(String.valueOf(cause.getState()))
-        ));
+        );
+        parameters.addAll(pluginParameters);
+
+        try {
+            Constructor<ParametersAction> constructor = ParametersAction.class.getConstructor(List.class, Collection.class);
+            Set<String> names = new HashSet<>();
+            for (ParameterValue param : pluginParameters) {
+                names.add(param.getName());
+            }
+            parametersAction = constructor.newInstance(parameters, names);
+        } catch (NoSuchMethodException | IllegalAccessException | InstantiationException
+                | InvocationTargetException ex) {
+            parametersAction = new ParametersAction(parameters);
+        }
+
         GitHubPRBadgeAction gitHubPRBadgeAction = new GitHubPRBadgeAction(cause);
 
         ParameterizedJobMixIn parameterizedJobMixIn = asParameterizedJobMixIn(job);
@@ -290,8 +309,12 @@ public class JobRunnerForCause implements Predicate<GitHubPRCause> {
             LOGGER.error("Couldn't extract quiet period, falling back to {}", quietPeriod, e);
         }
 
-        return parameterizedJobMixIn.scheduleBuild2(quietPeriod, new CauseAction(cause), new ParametersAction(values),
-                gitHubPRBadgeAction);
+        return parameterizedJobMixIn.scheduleBuild2(
+                quietPeriod,
+                new CauseAction(cause),
+                parametersAction,
+                gitHubPRBadgeAction
+        );
     }
 
     protected static class CausesFromAction implements Function<Action, Iterable<Cause>> {

